src/rqt_gauges_2/steering_wheel_gauge.py

```diff
@@ -0,0 +1,160 @@
+
+        super(SteeringWheelGauge, self).__init__(parent)
+
+        self.value = 0
+        self.width = 400
+        self.height = 400
+        self.progress_width = 25
+        self.progress_rounded_cap = True
+        self.progress_color = 0x39F030
+        self.max_value = 45
+        self.circle_max_angle = 135
+        self.font_size = 40
+        self.scale_font_size = 15
+        self.font_family = "Verdana"
+        self.suffix = "°"
+        self.text_color = 0x000000
+        self.enable_shadow = True
+
+        self.angle_offset = 0
+
+        self.minValue = -45
+        self.maxValue = 45
+
+        if self.width <= self.height:
+            self.widget_diameter = self.width
+        else:
+            self.widget_diameter = self.height
+
+        self.resize(self.width, self.height)
+
+
+    def draw_big_scaled_marker(self, start_angle_value, angle_size, scalaCount):
+        painter = QPainter(self)
+        painter.setRenderHint(QPainter.Antialiasing)
+        painter.translate(self.width / 2, self.height / 2)
+
+        self.pen = QPen(Qt.black)
+        self.pen.setWidth(2)
+        painter.setPen(self.pen)
+
+        painter.rotate(start_angle_value - self.angle_offset)
+        steps_size = (float(angle_size) / float(scalaCount))
+        scale_line_outer_start = (self.widget_diameter)/2
+        scale_line_lenght = (self.widget_diameter / 2) - (self.widget_diameter / 20)
+
+        for i in range(scalaCount+1):
+            painter.drawLine(scale_line_lenght, 0, scale_line_outer_start, 0)
+            painter.rotate(steps_size)
+        painter.end()
+
+    def create_scale_marker_values_text(self, scale_angle_start_value, scale_angle_size, scalaCount):
+        painter = QPainter(self)
+        painter.setRenderHint(QPainter.TextAntialiasing)
+        painter.translate(self.width / 2, self.height / 2)
+        font = QFont(self.font_family, self.scale_font_size, QFont.Bold)
+        fm = QFontMetrics(font)
+
+        pen_shadow = QPen()
+
+        pen_shadow.setBrush(QColor(0, 0, 0, 255))
+        painter.setPen(pen_shadow)
+
+        text_radius_factor = 0.75
+        text_radius = self.widget_diameter/2 * text_radius_factor
+
+        scale_per_div = int((self.maxValue - self.minValue) / scalaCount)
+
+        angle_distance = (float(scale_angle_size) / float(scalaCount))
+        for i in range(scalaCount + 1):
+            text = str(int(self.minValue + scale_per_div * i))
+            w = fm.width(text) + 1
+            h = fm.height()
+            painter.setFont(QFont(self.font_family, self.scale_font_size))
+            angle = angle_distance * i + float(scale_angle_start_value - self.angle_offset)
+            x = text_radius * math.cos(math.radians(angle))
+            y = text_radius * math.sin(math.radians(angle))
+
+            text = [x - int(w/2), y - int(h/2), int(w), int(h), Qt.AlignCenter, text]
+            painter.drawText(text[0], text[1], text[2], text[3], text[4], text[5])
```

Comment:
Can we create a function and isolate a bit the functionality?

Reply:
Hmm we certainly can, however, it is using several local variables defined on the same method. Does it makes sense to create a function with all of those arguments that is only been used at that place in order to clean it up a little bit?